centreon/src/Core/Host/Application/UseCase/PartialUpdateHost/PartialUpdateHostValidation.php

```diff
@@ -0,0 +1,291 @@
+{
+    use LoggerTrait;
+
+    /**
+     * @param ReadHostRepositoryInterface $readHostRepository
+     * @param ReadMonitoringServerRepositoryInterface $readMonitoringServerRepository
+     * @param ReadHostTemplateRepositoryInterface $readHostTemplateRepository
+     * @param ReadViewImgRepositoryInterface $readViewImgRepository
+     * @param ReadTimePeriodRepositoryInterface $readTimePeriodRepository
+     * @param ReadHostSeverityRepositoryInterface $readHostSeverityRepository
+     * @param ReadTimezoneRepositoryInterface $readTimezoneRepository
+     * @param ReadCommandRepositoryInterface $readCommandRepository
+     * @param ReadHostCategoryRepositoryInterface $readHostCategoryRepository
+     * @param ReadHostGroupRepositoryInterface $readHostGroupRepository
+     * @param InheritanceManager $inheritanceManager
+     * @param ContactInterface $user
+     * @param AccessGroup[] $accessGroups
+     */
+    public function __construct(
+        private readonly ReadHostRepositoryInterface $readHostRepository,
+        private readonly ReadMonitoringServerRepositoryInterface $readMonitoringServerRepository,
+        private readonly ReadHostTemplateRepositoryInterface $readHostTemplateRepository,
+        private readonly ReadViewImgRepositoryInterface $readViewImgRepository,
+        private readonly ReadTimePeriodRepositoryInterface $readTimePeriodRepository,
+        private readonly ReadHostSeverityRepositoryInterface $readHostSeverityRepository,
+        private readonly ReadTimezoneRepositoryInterface $readTimezoneRepository,
+        private readonly ReadCommandRepositoryInterface $readCommandRepository,
+        private readonly ReadHostCategoryRepositoryInterface $readHostCategoryRepository,
+        private readonly ReadHostGroupRepositoryInterface $readHostGroupRepository,
+        private readonly InheritanceManager $inheritanceManager,
+        private readonly ContactInterface $user,
+        public array $accessGroups = [],
+    ) {
+    }
+
+    /**
+     * Assert name is not already used.
+     *
+     * @param string $name
+     * @param Host $host
+     *
+     * @throws HostException|\Throwable
+     */
+    public function assertIsValidName(string $name, Host $host): void
+    {
+        if ($host->isNameIdentical($name)) {
+
+            return;
+        }
+        $formattedName = Host::formatName($name);
+        if ($this->readHostRepository->existsByName($formattedName)) {
+            $this->error('Host name already exists', compact('name', 'formattedName'));
+
+            throw HostException::nameAlreadyExists($formattedName, $name);
+        }
+
+        if (str_starts_with($name, '_Module_')) {
+            throw HostException::nameIsInvalid();
+        }
+    }
+
+    /**
+     * Assert monitoring server ID is valid.
+     *
+     * @param int $monitoringServerId
+     *
+     * @throws HostException|\Throwable
+     */
+    public function assertIsValidMonitoringServer(int $monitoringServerId): void
```

Comment:
Apparent unused method `assertIsValidMonitoringServer`.
Can you check it out and eventualy remove it ?

Reply:
I forgot to call it 😞